### ISSUE TITLE
`from_str_radix` hex "+" pitfall

### ISSUE THREAD

**Issue by Rudxain** [rust-lang/rust-clippy#16213]
### Description

[This seems like a common mistake](https://github.com/search?q=u8%3A%3Afrom_str_radix%28&type=code). It consists on using code like this:
```rust
uN::from_str_radix(hex_slice, 16)
```
to decode a hexadecimal string into an integer, unintentionally (in most cases) [allowing `+` as a valid prefix](https://doc.rust-lang.org/1.91.1/std/primitive.u8.html#method.from_str_radix) for hex sequences. I say "unintentional" because hex-strs typically represent arbitrary data, not numbers/numerals.

> [!note]
> Prev paragraph was edited for clarity

IDK if there's an existing lint-rule where this should be included, or if it warrants a new lint, that's why I've opened a blank issue

### Version

```text

```

### Additional Labels

@rustbot label +C-enhancement

**Comment by profetia** [comment]
Isn't `from_str_radix` already return a `Result`?

**Comment by Jarcho** [comment]
`from_str_radix(x, 16)` allows strings like `+0` which is probably not what people are expecting going by that github search.
